### ISSUE TITLE
URLPath.click incorrectly adds extra slashes on python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8063]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#8063|
|Type|release blocker: regression|
|Created|2015-10-24 06:49:32Z|

python 2 (what I would expect):

```
$ python -c 'from twisted.python.urlpath import URLPath; print(str(URLPath.fromBytes(b"http://localhost/").click(b"/hello")))'
http://localhost/hello
```

python 3 (oops):

```
$ python -c 'from twisted.python.urlpath import URLPath; print(str(URLPath.fromBytes(b"http://localhost/").click(b"/hello")))'
http://localhost//hello
```

I am planning to fix this in the [ticket:5388 IRI ticket] since gutting `URLPath` will be a good way of validating that the new algorithms work properly for existing stuff, and the new `URL` object already has tests for edge cases like this, whereas `URLPath` itself has almost no coverage.

I am tagging this as a "regression" even though it isn't really, because if this functionality started behaving this way on py2 it _would_ be a regression.  So the port isn't really finished and it would be a bit misleading to ship with that in the NEWS file.  Someone else can feel free to disagree with me of course :).

<details><summary>Searchable metadata</summary>

```
trac-id__8063 8063
type__release_blocker__regression release blocker: regression
reporter__glyph glyph
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1445669372002521 1445669372002521
changetime__1445854184942426 1445854184942426
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Fix is up for review on [#5388](https://github.com/twisted/twisted/issues/5388).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set status to `closed`|
|-|-|

(In [45992]) Merge iri-5388-4: new, internationalizable URL

Author: glyph, itamar

Reviewer: jonathanj, hawkowl

Fixes: [#5388](https://github.com/twisted/twisted/issues/5388)
Fixes: [#8063](https://github.com/twisted/twisted/issues/8063)

twisted.python.url is a new abstraction for URLs, supporting RFC 3987
IRIs.

This change also replaces the guts of twisted.python.urlpath, reducing
duplication since we cannot deprecate it immediately.
